src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/settings/panels/SettingsPanelNotifications.tsx

```diff
@@ -2,8 +2,219 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-function SettingsPanelNotifications() {
-  return <div>Set notifications content</div>;
+import { SubscriberRow } from "knex/types/tables";
+import { SubscriberEmailPreferencesOutput } from "../../../../../../../../db/tables/subscriber_email_preferences";
+import { Session } from "next-auth";
+import { useL10n } from "../../../../../../../hooks/l10n";
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import { hasPremium } from "../../../../../../../functions/universal/user";
+import { createContext, useState } from "react";
+import { RadioGroupState, useRadioGroupState } from "react-stately";
+import {
+  EmailUpdateCommOptionRequest,
+  EmailUpdateCommTypeOfOptions,
+} from "../../../../../../../api/v1/user/update-comm-option/route";
+import { useTelemetry } from "../../../../../../../hooks/useTelemetry";
+import { useRadioGroup } from "react-aria";
+import { VisuallyHidden } from "../../../../../../../components/server/VisuallyHidden";
+import { Button } from "../../../../../../../components/client/Button";
+import styles from "./SettingsPanelNotifications.module.scss";
+import { SwitchInput } from "../../../../../../../components/client/SwitchInput";
+import { RadioInput } from "../../../../../../../components/client/RadioInput";
+
+export type SettingsPanelNotificationsProps = {
+  data: SubscriberEmailPreferencesOutput;
+  subscriber: SubscriberRow;
+  user: Session["user"];
+};
+
+export type NotificationSettingsProps = {
+  user: Session["user"];
+  subscriber: SubscriberRow;
+  data: SubscriberEmailPreferencesOutput;
+};
+
+const AlertAddressContext = createContext<RadioGroupState | null>(null);
+
+export const NotificationsSettings = (props: NotificationSettingsProps) => {
+  const l10n = useL10n();
+  const session = useSession();
+  const router = useRouter();
+
+  const breachAlertsEmailsAllowed = props.subscriber.all_emails_to_primary;
+
+  // Extract monthly report preference from the right column
+  const monitorReportAllowed = hasPremium(props.user)
+    ? props.data.monthly_monitor_report
+    : props.data.monthly_monitor_report_free;
+
+  const defaultActivateAlertEmail =
+    typeof breachAlertsEmailsAllowed === "boolean";
+  const [activateAlertEmail, setActivateAlertEmail] = useState<boolean>(
+    defaultActivateAlertEmail,
+  );
+
+  const [activateMonthlyMonitorReport, setActivateMonthlyMonitorReport] =
+    useState(monitorReportAllowed);
+
+  const commsValue = () => {
+    switch (breachAlertsEmailsAllowed) {
+      case true:
+        return "primary";
+      case false:
+        return "affected";
```

Comment:
I'd suggest using an enum or a constant here instead of a string, that way a typo in one place will be more detectable. You may alternatively be able to get TypeScript to check the string, I'm not sure how to do that here offhand though.

Reply:
Good point! I’ve updated this part in [5d5853c](https://github.com/mozilla/blurts-server/pull/5342/commits/5d5853ce14372c4ad1b41d4b0c92d0d4e03c985c). This was taken over from the current implementation in the `AlertAddressForm` component. I did not update it there, because the entire file will be deprecated after the settings redesign is live.